routers/api/v1/misc/nodeinfo.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package misc
+
+import (
+	"net/http"
+
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/structs"
+)
+
+// NodeInfo returns the NodeInfo for the Gitea instance to allow for federation
+func NodeInfo(ctx *context.APIContext) {
+	// swagger:operation GET /nodeinfo miscellaneous getNodeInfo
+	// ---
+	// summary: Returns the nodeinfo of the Gitea application
+	// produces:
+	// - application/json
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/NodeInfo"
+
+	nodeInfo := &structs.NodeInfo{
+		Version: "2.1",
+		Software: structs.NodeInfoSoftware{
+			Name:       "gitea",
+			Version:    setting.AppVer,
+			Repository: "https://github.com/go-gitea/gitea.git",
+			Homepage:   "https://gitea.io/",
+		},
+		Protocols: []string{"activitypub"},
```

Comment:
But we don't currently support activitypub do we?

Reply:
per the schema, this requires a minimum length of 1 (https://github.com/jhass/nodeinfo/blob/main/schemas/2.1/schema.json#L54), and I figured this PR is the start of activitypub so it's ok to include it.

Reply:
`git` until we have the first federation api?

Reply:
`git` is not a supported value in this field (check the link I provided for possible values)